src/components/vm/disks/diskAdd.jsx

```diff
@@ -513,11 +515,40 @@ export class AddDiskModalBody extends React.Component {
             break;
         }
         case 'file': {
+            this.setState({ file: value });
+
             if (value.endsWith(".iso")) {
                 // use onValueChange instead of setState in order to perform subsequent state change logic
                 this.onValueChanged("device", "cdrom");
             }
-            this.setState({ file: value });
+
+            if (value) {
+                cockpit.spawn(["head", "--bytes=16", value], { binary: true, err: "message", superuser: "try" })
+                        .then(file_header => {
+                            let format = "";
+
+                            if (file_header[0] == 81 && file_header[1] == 70 &&
+                                file_header[2] == 73 && file_header[3] == 251) {
+                                format = "qcow2";
+                                // All zeros, no backing file offset
+                                if (file_header.slice(8).every(bytes => bytes === 0)) {
+                                    this.setState({ customDiskVerified: true, format: format, dialogError: "" });
+                                } else {
+                                    this.dialogErrorSet(_("Importing an image with a backing file is unsupported"));
+                                    this.setState({ customDiskVerified: false });
+                                }
+                            } else {
+                                format = "raw";
+                                this.setState({ customDiskVerified: true, format: format, dialogError: "" });
+                            }
+                        })
+                        .catch(e => {
+                            this.setState({ customDiskVerified: true, format: "raw", dialogError: "" });
+                            console.warn("could not execute head --bytes=16", e.toString());
+                        });
+            } else {
+                this.setState({ customDiskVerified: false });
```

Comment:
Yeah as I described in this comment, this disables the "Add" button if "Custom path" field is empty, but doesn't provide any message why it's disabled.
![Screenshot from 2022-11-17 01-41-21](https://user-images.githubusercontent.com/42733240/202326661-13c994da-914a-451e-952c-408dc9fdf8b1.png)

We should maybe move this check into [validateParams()](https://github.com/cockpit-project/cockpit-machines/blob/main/src/components/vm/disks/diskAdd.jsx#L418) aim for the approach that we let user click on "Add" button, then [validateParams()](https://github.com/cockpit-project/cockpit-machines/blob/main/src/components/vm/disks/diskAdd.jsx#L418) is called, it checks that some compulsory input field is empty, and shows message, similar to this:
![Screenshot from 2022-11-17 01-46-58](https://user-images.githubusercontent.com/42733240/202326958-de6bb2a8-4f71-4c6f-9a5a-8aa2498d94f4.png)

Reply:
Why is the button missing here? Is this due to dark mode, where I changed the disabled state? (And do I need to write a workaround for the workaround?)

![image](https://user-images.githubusercontent.com/10246/202491410-9f01e3e4-48c3-402a-a451-019cee895339.png)

Reply:
@garrett What button is missing here? Do you mean that the "Add" button is disabled by default imediatelly when user opens the dialog? Yeah that was what my issue comment was about. It was disabled because the "Custom path" was empty, which I believe was incorrect.
But Jelle fixed it now, and now it disables button only when user inputs a path to a file which we cannot attach as a disk:
![Screenshot from 2022-11-22 14-51-12](https://user-images.githubusercontent.com/42733240/203331109-63ee679e-0cbd-44a4-93bd-0949b85ed9cb.png)

But when user doesn't input anything, it doesn't disable the button as before:
![Screenshot from 2022-11-22 14-51-22](https://user-images.githubusercontent.com/42733240/203331218-8a67b44e-f353-456b-abb1-82db1a5958c4.png)

Ideally we probably also want some form verification, which would check if the field is empty, and say someting like the Custom path should not be empty... But that can be done in different PR.

Reply:
No, the button is missing. There is no button, just text. It might be a dark theme artifact that the button and the background are the same color. I don't know if that's our fault or not. I need something reproducible locally to test.

Reply:
Ahhh right, I didn't realize that the disabled button is not the inline button, so it should be the outline around the text even when it's disabled... Great catch @garrett